scripts/tile_benchmark/benchmark.py

```diff
@@ -0,0 +1,365 @@
+        next_level = level + 1
+        next_arr = pos_list[next_level]
+
+        for x in range(2 ** level):
+            for y in range(2 ** level):
+                # calculate top corner of next arr
+                x_next = x * 2
+                y_next = y * 2
+                for ix in range(2):
+                    for iy in range(2):
+                        next_arr[y_next + iy, x_next + ix] = \
+                            np.core.defchararray.add(
+                                current_arr[y, x],
+                                base[iy,
+                                     ix])
+
+    data[ATLAS_ARR] = pos_list
+    return data
+
+
+def girder_init(host, image_id):
+    image_summary_url = "%sapi/v1/item/%s/tiles" % (host, image_id)
+    image_summary_req = requests.get(image_summary_url)
+
+    data = {}
+    if image_summary_req.status_code == requests.codes.ok:
+        try:
+            img_data = image_summary_req.json()
+
+            data[NUM_X_TILES] = math.ceil(img_data["sizeX"] /
+                                          img_data["tileWidth"])
+            data[NUM_Y_TILES] = math.ceil(img_data["sizeY"] /
+                                          img_data["tileHeight"])
+            data[URL] = "{host_name}api/v1/item/{img_num}/tiles/zxy/%d/%d/%d" \
+                .format(host_name=host, img_num=image_id)
+            data[NUM_LEVELS] = img_data["levels"]
+        except ValueError:
+            print" Error parsing image summary data from the following " \
+                 "request \n %s" % (image_summary_url)
+            sys.exit()
+    else:
+        print"Could not recieve meta data using the following request\n %s" \
+             % (image_summary_url)
+        sys.exit()
+
+    return data
+
+
+def girder_request(data, z, x, y):
+    # calc range of actual tiles
+    x_range = int(math.ceil(data[NUM_X_TILES] /
+                            2 ** ((data[NUM_LEVELS] - 1) - z)))
+    y_range = int(math.ceil(data[NUM_Y_TILES] /
+                            2 ** ((data[NUM_LEVELS] - 1) - z)))
+
+    if x >= x_range or y >= y_range:
+        return -1
+    tile_req = requests.get(data[URL] % (z, x, y))
+    elapsed_time = tile_req.elapsed.total_seconds()
+
+    if tile_req.status_code == requests.codes.ok:
+        # i = Image.open(StringIO(tile_req.content))
+        # i.show()
+        return elapsed_time
+
+    elif tile_req.status_code == 404:
+        print " 404 error to get tile from layer %d x %d y %d error code %d" \
+              % (z, x, y, tile_req.status_code)
+        return -1
+
+    elif tile_req.status_code != 404:
+        print " error to get tile from layer %d x %d y %d error code %d" \
+              % (z, x, y, tile_req.status_code)
+        return -1
+
+
+def atlas_request(data, z, x, y):
+    # print data[URL]%(data[ATLAS_ARR][z][y,x])
+    tile_req = requests.get(data[URL] % (data[ATLAS_ARR][z][y, x]))
+    elapsed_time = tile_req.elapsed.total_seconds()
```

Comment:
Let's always multiply this by `1000` as soon as we get it, so all of our times are in milliseconds.